prover/crates/lib/witness_generator_service/src/job_runner.rs

```diff
@@ -0,0 +1,49 @@
+use std::sync::Arc;
+
+use tokio_util::sync::CancellationToken;
+use zksync_object_store::ObjectStore;
+use zksync_prover_dal::{ConnectionPool, Prover};
+use zksync_prover_job_processor::{Backoff, BackoffAndCancellable, JobRunner};
+use zksync_types::protocol_version::ProtocolSemanticVersion;
+
+use crate::{
+    artifact_manager::ArtifactsManager,
+    executor::WitnessGeneratorExecutor,
+    job_picker::WitnessGeneratorJobPicker,
+    job_saver::WitnessGeneratorJobSaver,
+    rounds::{JobManager, VerificationKeyManager},
+};
+
+pub fn witness_generator_runner<R>(
+    max_circuits_in_flight: usize,
+    blob_store: Arc<dyn ObjectStore>,
+    pool: ConnectionPool<Prover>,
+    num_workers: usize,
+    protocol_version: ProtocolSemanticVersion,
+    keystore: Arc<dyn VerificationKeyManager>,
+    cancellation_token: CancellationToken,
```

Comment:
It's better to pass `cancellation_token` to `run` fn instead of storing it in a struct.